Subject: config.sub: accept substrate as an OS name

Add substrate to the recognised OS list so --host=i386-unknown-substrate
passes configure's host-triple validation.

--- a/config.sub
+++ b/config.sub
@@ -2108,7 +2108,7 @@
 	| skyos* \
 	| solaris* \
 	| solidbsd* \
-	| sortix* \
+	| sortix* | substrate* \
 	| storm-chaos* \
 	| sunos \
 	| sunos[34]* \
